plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt.go

```diff
@@ -50,13 +52,974 @@ type CiscoTelemetryMDT struct {
+	c.nxpathMap[key]["input_type"] = "string"
+	c.nxpathMap[key]["allocated"] = "string"
+	c.nxpathMap[key]["fanhwver"] = "string"
+	c.nxpathMap[key]["clkhwver"] = "string"
+	c.nxpathMap[key]["fannum"] = "string"
+	c.nxpathMap[key]["watts_requested"] = "string"
+	c.nxpathMap[key]["cumulative_power"] = "string"
+	c.nxpathMap[key]["tot_gridB_capacity"] = "string"
+	c.nxpathMap[key]["pow_used_by_mods"] = "string"
+	c.nxpathMap[key]["tot_pow_alloc_budgeted"] = "string"
+	c.nxpathMap[key]["psumod"] = "string"
+	c.nxpathMap[key]["ps_status_3k"] = "string"
+	c.nxpathMap[key]["temptype"] = "string"
+	c.nxpathMap[key]["regval"] = "string"
+	c.nxpathMap[key]["inlet_temp"] = "string"
+	c.nxpathMap[key]["det_cord"] = "string"
+	c.nxpathMap[key]["reserve_fan"] = "string"
+	c.nxpathMap[key]["det_pina"] = "string"
+	c.nxpathMap[key]["minthres"] = "string"
+	c.nxpathMap[key]["actual_draw"] = "string"
+	c.nxpathMap[key]["sensor"] = "string"
+	c.nxpathMap[key]["zone"] = "string"
+	c.nxpathMap[key]["det_iin"] = "string"
+	c.nxpathMap[key]["det_iout"] = "string"
+	c.nxpathMap[key]["det_vin"] = "string"
+	c.nxpathMap[key]["fan1rpm"] = "string"
+	c.nxpathMap[key]["tot_gridA_capacity"] = "string"
+	c.nxpathMap[key]["fanperc"] = "string"
+	c.nxpathMap[key]["det_pout"] = "string"
+	c.nxpathMap[key]["alarm_str"] = "string"
+	c.nxpathMap[key]["zonespeed"] = "string"
+	c.nxpathMap[key]["det_total_cap"] = "string"
+	c.nxpathMap[key]["reserve_xbar"] = "string"
+	c.nxpathMap[key]["det_vout"] = "string"
+	c.nxpathMap[key]["watts_alloced"] = "string"
+	c.nxpathMap[key]["ps_in_power"] = "string"
+	c.nxpathMap[key]["tot_pow_input_actual_draw"] = "string"
+	c.nxpathMap[key]["ps_output_voltage"] = "string"
+	c.nxpathMap[key]["det_name"] = "string"
+	c.nxpathMap[key]["tempmod"] = "string"
+	c.nxpathMap[key]["clockname"] = "string"
+	c.nxpathMap[key]["fanname"] = "string"
+	c.nxpathMap[key]["regnumstr"] = "string"
+	c.nxpathMap[key]["bitnumstr"] = "string"
+	c.nxpathMap[key]["ps_slot"] = "string"
+	c.nxpathMap[key]["actual_out"] = "string"
+	c.nxpathMap[key]["ps_input_voltage"] = "string"
+	c.nxpathMap[key]["psmodel"] = "string"
+	c.nxpathMap[key]["speed"] = "string"
+	c.nxpathMap[key]["clkmodel"] = "string"
+	c.nxpathMap[key]["ps_redun_mode_3k"] = "string"
+	c.nxpathMap[key]["tot_pow_capacity"] = "string"
+	c.nxpathMap[key]["amps"] = "string"
+	c.nxpathMap[key]["available_pow"] = "string"
+	c.nxpathMap[key]["reserve_supxbarfan"] = "string"
+	c.nxpathMap[key]["watts"] = "string"
+	c.nxpathMap[key]["det_pinb"] = "string"
+	c.nxpathMap[key]["det_vinb"] = "string"
+	c.nxpathMap[key]["ps_state"] = "string"
+	c.nxpathMap[key]["det_sw_alarm"] = "string"
+	c.nxpathMap[key]["regnum"] = "string"
+	c.nxpathMap[key]["amps_requested"] = "string"
+	c.nxpathMap[key]["fanrpm"] = "string"
+	c.nxpathMap[key]["actual_input"] = "string"
+	c.nxpathMap[key]["outlet_temp"] = "string"
+	c.nxpathMap[key]["tot_capa"] = "string"
+
+	return nil
+}
+
+func (c *CiscoTelemetryMDT) InitMemPhys() error {
```

Comment:
Please make this function private.

Reply:
All these functions are updating nxpathMap that's why they are defined with
func (c *CiscoTelemetryMDT) InitMemPhys()
c.nxpathMap[key]["processname"] = "string"

Reply:
Did you mean following changes to make function private 
-func (c *CiscoTelemetryMDT) InitProc() error {
+func InitProc(c *CiscoTelemetryMDT) {

Reply:
No I want to make it `func (c *CiscoTelemetryMDT) initProc() error` do it will definitively stay inside of this package. In case a function starts with lower-case it will not be visible from outside the package.